frontend/fingertips-frontend/components/forms/SearchForm/searchActions.ts

```diff
@@ -0,0 +1,42 @@
+'use server';
+
+import { z } from 'zod';
+import { redirect } from 'next/navigation';
+
+const $SearchFormSchema = z.object({
+  indicator: z.string({
+    invalid_type_error: 'Please enter an indicator id',
+  }),
+});
+
+export type State = {
+  errors?: {
+    indicator?: string[];
+  };
+  message?: string | null;
+};
+
+export type SearchForm = z.infer<typeof $SearchFormSchema>;
+
+export type SearchFormState<T = SearchForm> = State & T;
+
+export async function searchIndicator(
+  prevState: SearchFormState,
+  formData: FormData
+) {
+  const validatedFields = $SearchFormSchema.safeParse({
+    indicator: formData.get('indicator'),
+  });
+
+  if (!validatedFields.success) {
+    return {
+      indicator: formData.get('indicator')?.toString() ?? '',
+      errors: validatedFields.error.flatten().fieldErrors,
+      message: 'Missing Fields',
+    };
+  }
```

Comment:
Is this validation working? Even if I don't enter a value it lets me go to the results page.

Reply:
Nope not working. Taken from an example but this can be built upon for the actual ticket.

Reply:
OK, happy to resolve this for now then.